pandas/tests/frame/methods/test_interpolate.py

```diff
@@ -88,7 +85,8 @@ def test_interp_basic(self):
 
         # check we DID operate inplace
         assert np.shares_memory(df["C"]._values, cvalues)
-        assert np.shares_memory(df["D"]._values, dvalues)
+        if not using_infer_string:
+            assert np.shares_memory(df["D"]._values, dvalues)
```

Comment:
With EAs, I don't think there can be any expectation that this is holds.

Reply:
```suggestion
        assert tm.shares_memory(df["D"]._values, dvalues)
```

If we use our own helper that can handle EAs, it actually still appears to be true